querymsc.py

```diff
@@ -151,77 +151,56 @@ def query_annual_precip(self, session):
+                connection string and engine
+            percentile (float): desire percentile in fraction
         Returns: 
             query (sqlalchemy query): sqlalchemy query constructed 
                 using ORM to query temperature percentiles 
         """
 
-        # construct query table
+        # construct query
         query = (
-                session.query(
-                                (func.percentile_cont(0.01)
+                 session.query(
+                                (func.percentile_cont(percentile)
                                      .within_group(Obs.datum.asc())
                                      .label("temp")),
-                                 func.min(Obs.time).label("time_min"),
-                                 func.max(Obs.time).label("time_max"),
-                                 History.lat,
-                                 History.lon,
-                                 History.station_id)
-                            # select from and make join explicit
-                            .select_from([History, Variable])
-                            .filter(
-                                    and_(Obs.time >= self.start_time,
-                                         Obs.time <= self.end_time)
-                                    )
-                            .filter(
-                                    func.extract("month", Obs.time) == self.month
-                                    )
-                            .filter(
-                                    and_(Variable.standard_name == 'air_temperature',
-                                         Variable.id == 1510)
-                                    )
-                            .group_by(History.lat, 
-                                      History.lon, 
-                                      History.station_id)
+                                func.min(Obs.time).label("time_min"),
+                                func.max(Obs.time).label("time_max"),
+                                History.lat.label("lat"),
+                                History.lon.label("lon"),
+                                History.station_id.label("station_id"),
+                                func.count(Obs.datum).label('obs_count')
+                                 )
+                        .join(History)
+                        .join(Variable, Variable.id == Variable.id)
+                        .filter(
+                                and_(Obs.time >= self.start_time,
+                                     Obs.time <= self.end_time)
+                                     )
+                        .filter(
+                                func.extract("month", Obs.time) == self.month
+                                )
+                        .filter(
+                                and_(Variable.standard_name == 'air_temperature',
+                                     Variable.id == 1510)
```

Comment:
I'd think that `Variable.standard_name` is redundant to `Variable.id`, in which case I'd omit it.

Also suggest you document why you are specifying `Variable.id`; you mentioned this in meeting this morning, I believe, but it won't be obvious to a reader 6 months from now.

Reply:
Yes - ironically, I included the Variable.standard_name as a reminder to myself as to what each Variable.id resembled, but that wouldn't be clear to future users. Hopefully a unique and standard variable filtering method can be made in dbmsc.

Reply:
This is a natural place for parameterization. You certainly don't want to embed magic constants like `1510` in the code here.